assets/model_monitoring/components/tests/unit/test_gsq_input_schema_adaptor.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""test class for GSQ - input schema adaptor component."""
+
+import pytest
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructType, StructField, StringType
+from src.generation_safety_quality.input_schema_adaptor.run import (
+    _adapt_input_data_schema,
+)
+
+
+@pytest.mark.gsq_test
+@pytest.mark.unit
+class TestInputSchemaAdaptor:
+    """Test class for GSQ - Input Schema Adaptor."""
+
+    def _init_spark(self) -> SparkSession:
+        """Create spark session."""
+        return SparkSession.builder.appName("test").getOrCreate()
+
+    _expected_gsq_input_schema = StructType(
+        [
+            StructField("trace_id", StringType(), True),
+            StructField("context", StringType(), True),
+            StructField("groundtruth", StringType(), True),
+            StructField("prompt", StringType(), True),
+            StructField("output", StringType(), True),
+            StructField("root_span", StringType(), True),
+        ]
+    )
+
+    _expected_gsq_input_schema_extra = StructType(
+        [
+            StructField("trace_id", StringType(), True),
+            StructField("context", StringType(), True),
+            StructField("groundtruth", StringType(), True),
+            StructField("prompt", StringType(), True),
+            StructField("output", StringType(), True),
+            StructField("source", StringType(), True),
```

Comment:
what does this source mean? Sometimes the output has 2 parameters inside it?

Reply:
source was just a randomly named extra key that I included in the data to test behavior when there are extra data in the input/output dicts.

Reply:
So basically we want to expand all the fields in "input" and "output" as an individual column in gsq input? And the groundtruth, context, .. those information will be part of the "input"? Is my understanding correct?

Reply:
Yes that's correct. We cannot know the 'input' and 'output' schema in the adaptor that is up to the user we just expand and pass on to the gsq where they give us the column mapping.